python/cudf/cudf/core/frame.py

```diff
@@ -238,6 +238,17 @@ def _quantiles(
         result._copy_categories(self)
         return result
 
+    def shift(self, periods=1, freq=None, axis=0, fill_value=None):
+        """Shift values by `periods` positions.
+        """
+        assert axis in (None, 0) and freq is None
+        return self._shift(periods)
+
+    def _shift(self, offset, fill_value=None):
+        data_columns = (col.shift(offset, fill_value) for col in self._columns)
```

Comment:
`fill_value` needs to be converted into a Scalar at some point either here or in `copying.pyx`. I don't think it's reasonable to expect an end user to construct the Scalar themself.